modules/service/src/main/scala/lucuma/odb/graphql/GraphQLRoutes.scala

```diff
@@ -52,6 +54,9 @@ object GraphQLRoutes {
                   {
                     for {
                       user <- OptionT(client.get(a))
+                      // If the user has never hit the ODB using http then there will be no user
+                      // entry in the database. So go ahead and [re]canonicalize here to be sure.
+                      _    <- OptionT.liftF(userSvc.canonicalizeUser(user))
```

Comment:
I ran into this issue when we tested with Explore, which never hits the server via plain HTTP and thus never went through the normal user-canonicalizing code.